### PR TITLE
Improve how vending machine restocking works

### LINKED ISSUE #37581: Using RPED on vending machines deletes their stock
Round ID:87233

Test merged 

Server revision compiled on: 2018-04-29

The following pull requests are currently test merged:
#37504: 'Removes an unnecessary file causing github security warnings on tgui, also updates/downgrades some dependencies to ensure a build' by vuonojenmustaturska at commit ef8d06
#37410: '[syou can no longer insert a storage object into itself, and therefore, no longer bohbomb with one boh' by kevinz000 at commit 861dc5
#34086: 'goonchat linkify has been vastly improved' by optimumtact at commit cd19a3
#37406: 'Adds (fake) ambient occlusion as a client preference' by deathride58 at commit 87a9b1

Based off origin/master commit: eb5be5

Revision: 041fa8141b0b568bf018cdb1ebd0f08b55ee8333

Server tools version: /tg/station 13 Server v3.2.2.0

Server tools API version: 3.2.0.2

Image showing product and frontend view

Steps to reproduce unknown

Spawning a new boozemat did not have any issues

### LINKED ISSUE #37728: You can get infinite ammounts of vendor products.
This isn't really a big issue unless it works with donksoft vendors really, but those already have lots of stock in them so not even then.

Anyways, the thing goes like this:
1. hit vendor until products pop out, count them (they should be below the ammount you could get from emptying the vendor manually).
2. keep hitting till it breaks into generic machine frame.
3. rebuild with the stuff it dropped (restock modules).
4. repeat infinitely, getting products beyond what the stock on the machine says.

I didn't report this when it happened cause I'm a dumb dumb, the station was power sinked, if that's relevant at all.

### PR DESCRIPTION
**Pull request by SpaceManiac**
:cl:
tweak: One canister is now suitable to fully stock a vending machine. Relevant cargo packs have been updated.
fix: Deconstructing and reconstructing a vending machine no longer creates items from thin air.
fix: Using a part replacer on a vending machine no longer empties its inventory.
add: Bluespace RPEDs can now be used to refill vending machines.
/:cl:

Fixes #37581. Fixes #37728.

The "charges" system has been removed in favor of tracking exactly how many of which items are available on the restock units. A brand-new restock unit is "sealed", and will take on the product list of the first vending machine it is successfully used on, including during construction. The canister's amounts are updated during deconstruction. Bluespace RPEDs can now hold the BULKY restock units, and part replacement uses the restock logic rather than just swapping out canisters.

A good demonstration case is deconstructing and reconstructing a maintenance Booze-o-Mat. When deconstructed and reconstructed, it takes on the product list of the normal Booze-o-Mat, but the actual stock has been persisted via the restock unit.

Also fixes vending machines constructed while unanchored having inaccessible maintenance panels, by requiring them to be built anchored.

Some grammar and formatting fixes rolled in.

I changed the annoyingly hard-to-read dark blue color in the interface, and the others to match:

image

### CODE REVIEW

**Comment by Vidiveni** [comment]
List of items is still unsorted also.

**Comment by 81Denton** [comment]
Great PR! I remember that vending machines need three supply packs to be constructed; should that also be reduced to a single pack?

**Comment by zaracka** [comment]
muh realism

**Comment by Tlaltecuhtli** [comment]
can you also change the "screwdrive and then unwrench to move", all other meachines just need to be hit with a wrench to move

**Comment by zaracka** [comment]
I appreciate realism. Thank you for this.

File: code/modules/cargo/packs.dm [hunk 1]
```diff
@@ -1366,38 +1361,30 @@
 	name = "Cigarette Supply Crate"
 	desc = "Don't believe the reports - smoke today! Contains cigarette vending machine refills."
```
Comment (Mark9013100) [inline code/modules/cargo/packs.dm]: should be singular

**Comment by Anonmare** [comment]
reeee no more tacticool knock-off shop

**Comment by Bawhoppen** [comment]
Finally! This has been a long time coming.

**Comment by ExcessiveUseOfCobblestone** [comment]
Next goal is to modularize vending machines that _intake_ items.

Did you know every vending machine has a variable called dish something that is never used except for the snack machine? Isn't that wild?!!!

**Comment by epochayur** [comment]
+1 just for the text recolor

**Comment by Cyberboss** [comment]
Travis did suceed

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**